### PR TITLE
feat(azure/vm): add new check `vm_ensure_using_approved_images `

### PR DESCRIPTION
**Pull request by puchy22**
### Context

This PR adds a new Azure VM check: `vm_ensure_using_approved_images `.
It ensures that all Azure virtual machines are launched from approved (custom) machine images, supporting compliance and security best practices.

### Description

- Adds the `vm_ensure_using_approved_images ` check for the Azure provider.
- Updates the VM service model to include the `image_reference` attribute, enabling the check to determine the image used by each VM.
- Provides comprehensive unit tests for all logic branches and edge cases.
- Updates the changelog with the new check entry.

### Checklist

- Are there new checks included in this PR? **Yes**
    - If so, do we need to update permissions for the provider? No new dependencies are required for this change.
      **No additional permissions required for this check.**
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.
      _Not needed for this feature._
- [x] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
      _No README changes required._
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### API
- [ ] Verify if API specs need to be regenerated.
      _Not applicable._
- [ ] Check if version updates are required (e.g., specs, Poetry, etc.).
      _Not applicable._
- [ ] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable.
      _Not applicable._

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/vm/vm_image_approved_in_use/vm_image_approved_in_use.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "azure",
+  "CheckID": "vm_image_approved_in_use",
+  "CheckTitle": "Only approved machine images are in use",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/vm/vm_image_approved_in_use/vm_image_approved_in_use.metadata.json]:
```suggestion
  "CheckTitle": "Ensure that Azure VMs are using an approved machine image",
```

File: prowler/providers/azure/services/vm/vm_image_approved_in_use/vm_image_approved_in_use.metadata.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "azure",
+  "CheckID": "vm_image_approved_in_use",
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/vm/vm_image_approved_in_use/vm_image_approved_in_use.metadata.json]:
```suggestion
  "CheckID": "vm_ensure_using_approved_images",
```

**Review by MrCloudSec — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]
